rakelib/artifacts.rake

```diff
@@ -625,11 +659,22 @@ namespace "artifact" do
     out.name = oss ? "logstash-oss" : "logstash"
     out.architecture = "all"
     if bundle_jdk
-      case platform
-        when "redhat", "centos"
-          out.architecture = "x86_64"
-        when "debian", "ubuntu"
-          out.architecture = "amd64"
+      if jdk_arch == 'x86_64'
+        case platform
+          when "redhat", "centos"
+            out.architecture = "x86_64"
+          when "debian", "ubuntu"
+            out.architecture = "amd64"
+        end
+      elsif jdk_arch == 'arm64'
```

Comment:
Should `arch_suffix` be `out.architecture`? Given that this code is in two places, we might want to factor the architecture "normalization" out into a separate method

Reply:
we can't do `out.architecture = arch_suffix` because is case of `bundle_jdk` is false `arch_suffix` has to be `no-jdk` while `out.architecture` `all`. However I extract the rest of the condition also becase in the "out.architeture" ifs there is a bug of missed assignment